services/brig/src/Brig/App.hs

```diff
@@ -609,10 +626,19 @@ instance MonadIndexIO (AppT r) => MonadIndexIO (ExceptT err (AppT r)) where
 instance Monad m => HasRequestId (AppT r) where
   getRequestId = view requestId
 
-runAppT :: Env -> AppT '[Final IO] a -> IO a
-runAppT e (AppT ma) = runFinal $ runReaderT ma e
-
-runAppResourceT :: ResourceT (AppT '[Final IO]) a -> (AppT '[Final IO]) a
+runAppT :: Env -> AppT BrigCanonicalEffects a -> IO a
+runAppT e (AppT ma) =
+  runFinal
+    . embedToFinal
+    . interpretClientToIO (_casClient e)
+    . codeStoreToCassandra @Cas.Client
+    . brigTimeToMonadIO @IO (_currentTime e)
+    . passwordResetStoreToCodeStore
+    $ runReaderT ma e
+
+runAppResourceT ::
```

Comment:
This function is getting more dangerous as we add more effects. Right now we're fine, but any sort of resumptive effects will silently get the wrong results when being pushed through this.

Reply:
Are you referring to things like throwing and catching, e.g, like in `wrapMonadClient` in one of your first Spar PRs? https://github.com/wireapp/wire-server/pull/1775/files#diff-718f7e7fa3d155d592d1551bd8fb924edf14990399056f4696b90ca27933dde2R215

Reply:
An effect like `Error e` or `State s` will get the wrong result if it's pushed through `runAppResourceT`